udf/udf_json_extract_keyed_histogram_js.sql

```diff
@@ -0,0 +1,58 @@
+/*
+
+Returns an array of parsed structs from a JSON string representing a keyed histogram.
+
+*/
+
+CREATE TEMP FUNCTION
+  udf_json_extract_keyed_histogram_js (input STRING)
+  RETURNS ARRAY<STRUCT<key STRING,
+  bucket_count INT64,
+  histogram_type INT64,
+  `sum` INT64,
+  `range` ARRAY<INT64>,
+  `values` ARRAY<STRUCT<key INT64,
+  value INT64>> > >
+  LANGUAGE js AS """
+    if (input == null) {
+      return null;
+    }
+    var result = [];
+    var parsed = JSON.parse(input);
+    for (var histoKey in parsed) {
+      var histogram = parsed[histoKey]
+      var valuesMap = histogram.values;
+      var valuesArray = [];
+      for (var key in valuesMap) {
+        valuesArray.push({"key": parseInt(key), "value": valuesMap[key]})
+      }
+      histogram.values = valuesArray;
+      histogram.key = histoKey
+      result.push(histogram)
+    }
+    result.values = valuesArray;
```

Comment:
Is there a reason why an attribute `values` is added to the `result` array containing the histogram values of only the last element? 
Not sure if it matters here, but if the JSON string was empty, then `valuesArray` would be undefined since it is first initialized in the `for` loop.

Reply:
Good catch. This line is a holdover from the non-keyed version of the function. There is no top-level `values` field in the returned SQL type, so the value set here is probably dropped as the result is transformed to SQL type. I will remove this line.